offchain-modules/src/packages/handlers/ckb.ts

```diff
@@ -1,51 +1,363 @@
 
 // CKB handler
 // 1. Listen CKB chain to get new burn events.
 // 2. Listen database to get new mint events, send tx.
 export class CkbHandler {
+  private ckb = ForceBridgeCore.ckb;
+  private indexer = ForceBridgeCore.indexer;
+  private PRI_KEY = ForceBridgeCore.config.ckb.privateKey;
   constructor(private db: CkbDb) {}
 
   // save unlock event first and then
-  async saveBurnEvent(burn: CkbBurn): Promise<void> {
-    logger.debug('save burn event');
-    // const unlock = await transformBurnEvent(burn);
-    // switch (unlock.name) {
-    //   case 'EthUnlock': {
-    //     await this.db.createEthUnlock([unlock]);
-    //     break;
-    //   }
-    //   default: {
-    //     throw new Error(`wrong unlock type: ${unlock.name}`);
-    //   }
-    // }
-    // await this.db.saveCkbBurn([burn]);
+  async saveBurnEvent(burns: ICkbBurn[]): Promise<void> {
+    logger.debug('save burn event:', burns);
+    for (const burn of burns) {
+      switch (burn.chain) {
+        case ChainType.ETH:
+          await this.db.createEthUnlock([
+            {
+              ckbTxHash: burn.ckbTxHash,
+              asset: burn.asset,
+              amount: burn.amount,
+              recipientAddress: burn.recipientAddress,
+            },
+          ]);
+          break;
+        case ChainType.TRON:
+          await this.db.createTronUnlock([
+            {
+              ckbTxHash: burn.ckbTxHash,
+              asset: burn.asset,
+              assetType: burn.asset,
+              amount: burn.amount,
+              recipientAddress: burn.recipientAddress,
+            },
+          ]);
+          break;
+        case ChainType.EOS:
+          await this.db.createEosUnlock([
+            {
+              ckbTxHash: burn.ckbTxHash,
+              asset: burn.asset,
+              amount: burn.amount,
+              recipientAddress: burn.recipientAddress,
+            },
+          ]);
+          break;
+        default:
+          throw new Error(`wrong burn chain type: ${burn.chain}`);
+      }
+      await this.db.createCkbBurn([burn]);
+    }
   }
 
   async watchBurnEvents(): Promise<never> {
     // get cursor from db, usually the block height, to start the poll or subscribe
     // invoke saveBurnEvent when get new one
+    let latestHeight = await this.db.getCkbLatestHeight();
+    while (true) {
+      logger.debug('watch burn event height: ', latestHeight);
+      const block = await this.ckb.rpc.getBlockByNumber(BigInt(latestHeight));
+      if (block == null) {
+        logger.debug('waitting for new ckb block');
+        await asyncSleep(5000);
+        continue;
+      }
+      const burnTxs = new Map();
+      for (const tx of block.transactions) {
+        const recipientData = tx.outputsData[0];
+        logger.debug('recipientData:', recipientData);
+        let cellData;
+        try {
+          cellData = new RecipientCellData(fromHexString(recipientData).buffer);
+        } catch (e) {
+          continue;
```

Comment:
should log error

Reply:
This is not an error scenario, no need to print the error log